sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3498,13 +3500,15 @@ object ArrayUnion {
   since = "2.4.0")
 case class ArrayIntersect(left: Expression, right: Expression) extends ArrayBinaryLike
   with ComplexTypeMergingExpression {
-  override def dataType: DataType = {
-    dataTypeCheck
```

Comment:
After checking the code, seems `dateTypeCheck` is not put in `checkInputDataTypes`

Reply:
Do you mean add it back?